analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -43,296 +56,278 @@ import org.ossreviewtoolkit.model.Scope
-            registrationNode["packageContent"].textValueOrEmpty()
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class MetaData(
+        // See https://docs.microsoft.com/en-us/nuget/reference/nuspec#required-metadata-elements
+        val id: String,
+        val version: String,
+        val description: String,
+        val authors: String,
+
+        // See https://docs.microsoft.com/en-us/nuget/reference/nuspec#optional-metadata-elements.
+        val licenseUrl: String? = null,
+        val license: License? = null,
+        val repository: Repository? = null,
+    )
 
-        private fun extractVersion(range: String): String {
-            if (range.isEmpty()) return ""
-            val rangeReplaces = range.replace("[", "")
-                .replace(" ", "")
-                .replace(")", "")
-            return rangeReplaces.split(",").elementAt(0)
-        }
-    }
+    data class License(
+        @JacksonXmlProperty(isAttribute = true)
+        val type: String,
+        @JacksonXmlProperty(isAttribute = true)
+        val version: String? = null,
 
-    val packages = sortedSetOf<Package>()
-    val issues = mutableListOf<OrtIssue>()
-    val scope = Scope("dependencies", sortedSetOf())
+        val value: String
+    )
 
-    // Maps an (id, version) pair to a (nupkg URL, catalog entry) pair.
-    private val packageReferencesAlreadyFound = mutableMapOf<Pair<String, String>, Pair<String, JsonNode>>()
+    data class Repository(
+        @JacksonXmlProperty(isAttribute = true)
+        val type: String? = null,
+        @JacksonXmlProperty(isAttribute = true)
+        val url: String? = null,
+        @JacksonXmlProperty(isAttribute = true)
+        val branch: String? = null,
+        @JacksonXmlProperty(isAttribute = true)
+        val commit: String? = null
+    )
 
-    init {
-        packageReferences.forEach { id ->
-            val scopeDependency = getPackageReferenceFromRestAPI(id.name, id.version)
-            scopeDependency?.let { scope.dependencies += it }
-        }
+    data class AllPackageData(
+        val data: PackageData,
+        val details: PackageDetails,
+        val spec: PackageSpec
+    )
 
-        scope.collectDependencies().forEach { id ->
-            val pkg = getPackageForId(id)
+    val jsonMapper = JsonMapper().registerKotlinModule()
+    val xmlMapper = XmlMapper(XmlFactory().apply {
+        // Work-around for https://github.com/FasterXML/jackson-module-kotlin/issues/138.
+        xmlTextElementName = "value"
+    }).registerKotlinModule()
 
-            if (pkg != Package.EMPTY) {
-                packages += pkg
-            }
-        }
-    }
+    private val serviceIndex = mapFromUrl<ServiceIndex>(jsonMapper, SERVICE_INDEX_URL)
 
-    private fun getPackageForId(packageId: Identifier) =
-        jsonNodeToPackage(getPackageJsonContent(packageId))
+    // Note: Remove a trailing slash to be able to use manually insert one before the path, as a double-slash would
```

Comment:
Something missing at around "insert"?   "be able to use manually insert one before the path"

Reply:
I've fixed the wording.